roles/openshift_health_checker/openshift_checks/efk_stack_check.py

```diff
@@ -0,0 +1,268 @@
+#!/usr/bin/env python
+'''
+  Send Cluster EFK checks to Zagg
+'''
+# vim: expandtab:tabstop=4:shiftwidth=4
+#
+#   Copyright 2015 Red Hat Inc.
+#
+#   Licensed under the Apache License, Version 2.0 (the "License");
+#   you may not use this file except in compliance with the License.
+#   You may obtain a copy of the License at
+#
+#       http://www.apache.org/licenses/LICENSE-2.0
+#
+#   Unless required by applicable law or agreed to in writing, software
+#   distributed under the License is distributed on an "AS IS" BASIS,
+#   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# This is not a module, but pylint thinks it is.  This is a command.
+# pylint: disable=invalid-name
+# If a check throws an exception it is failed and should alert
+# pylint: disable=bare-except
+import ssl
+import urllib2
+import json
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EFKStackCheck(OpenShiftCheck):
+    """"
+        This is a check for the entire EFK stack shipped with OCP
+        https://github.com/openshift/openshift-tools/blob/prod/scripts/monitoring/cron-send-logging-checks.py
+    """
+
+    name = "efk_stack_check"
+    tags = ["preflight", "elasticsearch", "efkstack"]
+
+    namespace = "logging"
+    config_file = "/etc/origin/master/admin.kubeconfig"
+
+    args = None
+    es_pods = []
+    fluentd_pods = []
+
+    def __init__(self, module_executor):
+        self.module_executor = module_executor
+
+        es_cert = '/etc/elasticsearch/secret/admin-'
+        self.es_curl = "curl -s --cert {}cert --key {}key --cacert {}ca -XGET ".format(es_cert, es_cert, es_cert)
+
+    def run(self, _, task_vars):
+        self.get_pods(task_vars)
+
+        status = dict()
+        status['elasticsearch'] = self.check_elasticsearch(task_vars)
+        status['fluentd'] = self.check_fluentd(task_vars)
+        status['kibana'] = self.check_kibana(task_vars)
+
+        return {"changed": True, "msg": status}
```

Comment:
@brenton was not sure what would constitute a "failure" for this check. In the case of no pods, routes, etc. present for elasticsearch, kibana, or fluentd, I receive an object that looks like: 

```json
{
    "elasticsearch": {
        "pods": {},
        "single_master": null
    },
    "fluentd": {
        "node_mismatch": 0,
        "number_expected_pods": 0,
        "number_pods": 0,
        "running": 1
    },
    "kibana": {}
}
```